prover/witness_generator/src/main.rs

```diff
@@ -125,7 +126,7 @@ async fn main() -> anyhow::Result<()> {
             .context("failed to build a prover_connection_pool")?;
     let (stop_sender, stop_receiver) = watch::channel(false);
 
-    let protocol_version = ProtocolVersionId::current_prover_version();
+    let protocol_version = ProtocolSemanticVersion::current_prover_version();
     let vk_commitments_in_db = match prover_connection_pool
         .connection()
         .await
```

Comment:
I wonder if we shouldn't move this block:

```
                let vk_commitments = get_cached_commitments();
                assert_eq!(
                    vk_commitments,
                    vk_commitments_in_db,
                    "VK commitments didn't match commitments from DB for protocol version {protocol_version:?}. Cached commitments: {vk_commitments:?}, commitments in database: {vk_commitments_in_db:?}"
                );
```

into here.
as it should probably trigger always, and not only in case of basic witness gen.

Reply:
well, that was the decision we made with Emil before because loading commitments takes around 40 seconds on every component startup, so we decided to do the check only in BWGs

Reply:
ah ok.